### ISSUE TITLE
Request for evaluation split/sample IDs for MindCube and VSI-Bench-tiny

### ISSUE THREAD

**Issue by 1-pluto1** [zhangzaibin/spagent#216]
Hi, thanks for your great work and for releasing the code!

I am trying to reproduce the experiments in the paper. In Section 4.1, the paper states that:

- BLINK (Multi-view) uses all multi-view data from the BLINK dataset.
- For MindCube, 40 questions are sampled from each of the three categories: rotation, around, and among, resulting in 120 evaluation questions.
- For VSI-Bench, the VSI-Bench-tiny split is used, and 7 frames are sampled from each video for evaluation.
- The RL training set contains 977 samples randomly selected from MindCube, with no overlap with the test set.

However, I could not find the exact evaluation split files or sample IDs for the MindCube subset and VSI-Bench-tiny subset in the repository.

Would it be possible to release the following files or provide instructions to reproduce them?

1. The MindCube evaluation sample IDs for the 120-question subset.
2. The 977 MindCube training sample IDs used for Think3D-RL.
3. The VSI-Bench-tiny sample/question/video IDs.
4. Any scripts used to generate these evaluation subsets.

These files would be very helpful for exact reproduction and fair comparison with the reported results.

Thank you very much!

**Comment by zhangzaibin** [comment]
Sorry for the late reply.

The dataset is available here: [[Think3DQA](https://huggingface.co/datasets/jialianjie/Think3DQA)](https://huggingface.co/datasets/jialianjie/Think3DQA).

For vLLM, we use the internal vLLM implementation integrated in Swift.

The MindCube evaluation subset can be found here:
https://drive.google.com/file/d/1_of11b2wYxRACno65mDyfT7AGVjv7hqL/view?usp=drive_link

For VSI-Bench, we use the official VSI-Bench-tiny split, which is available in the VSI-Bench repository.

**Comment by 1-pluto1** [comment]
Thank you for your reply.

I checked the VSI-Bench repository and HuggingFace dataset, but I still could not find the `VSI-Bench-tiny` split. I only found the full/debiased/pruned configurations.

In the VSI-Bench paper and project page, it seems that the tiny split is a randomly sampled subset for evaluation, but I could not find a publicly released file for this split.

Could you please provide the exact file path, commit, or link to the `VSI-Bench-tiny` split used in your experiments? If it is not publicly available, could you please share the sampled question IDs or the subset file used in Think3D?

Thank you very much!

**Comment by 1-pluto1** [comment]
Regarding vLLM, you mentioned that you used the internal vLLM implementation integrated in Swift. However, I could not find the specific Swift version or commit in the README or `requirements.txt`.

Could you please provide the Swift version/commit used in your experiments? If possible, could you also share the corresponding installation command or environment configuration for Swift?

**Comment by 1-pluto1** [comment]
> Sorry for the late reply.
> 
> The dataset is available here: [[Think3DQA](https://huggingface.co/datasets/jialianjie/Think3DQA)](https://huggingface.co/datasets/jialianjie/Think3DQA).
> 
> For vLLM, we use the internal vLLM implementation integrated in Swift.
> 
> The MindCube evaluation subset can be found here: https://drive.google.com/file/d/1_of11b2wYxRACno65mDyfT7AGVjv7hqL/view?usp=drive_link
> 
> For VSI-Bench, we use the official VSI-Bench-tiny split, which is available in the VSI-Bench repository.

could you please answer my new question?

**Comment by zhangzaibin** [comment]
Sorry for the late reply, I provide the pip list blow

absl-py                           2.3.1
accelerate                        1.10.1
addict                            2.4.0
aiofiles                          24.1.0
aiohappyeyeballs                  2.6.1
aiohttp                           3.13.1
aiosignal                         1.4.0
aliyun-python-sdk-core            2.16.0
aliyun-python-sdk-kms             2.16.5
annotated-types                   0.7.0
antlr4-python3-runtime            4.13.2
anyio                             4.11.0
astor                             0.8.1
asttokens                         3.0.0
attrdict                          2.0.1
attrs                             25.4.0
av                                16.0.1
binpacking                        1.5.2
blake3                            1.0.8
boto3                             1.40.55
botocore                          1.40.55
Brotli                            1.1.0
cachetools                        6.2.1
cbor2                             5.7.0
certifi                           2025.10.5
cffi                              2.0.0
charset-normalizer                3.4.4
click                             8.2.1
cloudpickle                       3.1.1
colorama                          0.4.6
colorlog                          6.10.1
compressed-tensors                0.11.0
contourpy                         1.3.3
cpm-kernels                       1.0.11
crcmod                            1.7
cryptography                      46.0.3
cupy-cuda12x                      13.6.0
cycler                            0.12.1
dacite                            1.9.2
datasets                          3.6.0
decorator                         5.2.1
decord                            0.6.0
deepspeed                         0.18.0
depyf                             0.19.0
dill                              0.3.8
diskcache                         5.6.3
distro                            1.9.0
dnspython                         2.8.0
docstring_parser                  0.17.0
dotenv                            0.9.9
einops                            0.8.1
email-validator                   2.3.0
et_xmlfile                        2.0.0
evalscope                         1.1.0
evaluate                          0.4.6
executing                         2.2.1
fastapi                           0.119.0
fastapi-cli                       0.0.13
fastapi-cloud-cli                 0.3.1
fastrlock                         0.8.3
ffmpy                             0.6.3
filelock                          3.20.0
fire                              0.7.1
fonttools                         4.60.1
frozendict                        2.4.6
frozenlist                        1.8.0
fsspec                            2025.3.0
func_timeout                      4.3.5
future                            1.0.0
fuzzywuzzy                        0.18.0
gguf                              0.17.1
google-auth                       2.41.1
google-genai                      1.45.0
gradio                            5.49.1
gradio_client                     1.13.3
groovy                            0.1.2
grpcio                            1.75.1
h11                               0.16.0
h5py                              3.15.1
hf-xet                            1.1.10
hjson                             3.1.0
httpcore                          1.0.9
httptools                         0.7.1
httpx                             0.28.1
huggingface-hub                   0.35.3
human-eval                        1.0.3
idna                              3.11
imageio                           2.37.0
immutabledict                     4.2.2
importlib_metadata                8.7.0
interegular                       0.3.3
ipdb                              0.13.13
ipython                           9.6.0
ipython_pygments_lexers           1.1.1
jedi                              0.19.2
jieba                             0.42.1
Jinja2                            3.1.6
jiter                             0.11.1
jmespath                          0.10.0
joblib                            1.5.2
json_repair                       0.52.1
json5                             0.12.1
jsonlines                         4.0.0
jsonschema                        4.25.1
jsonschema-specifications         2025.9.1
kiwisolver                        1.4.9
langdetect                        1.0.9
lark                              1.2.2
latex2sympy2_extended             1.0.6
Levenshtein                       0.27.1
llguidance                        0.7.30
llvmlite                          0.44.0
lm-format-enforcer                0.11.3
lxml                              6.0.2
Markdown                          3.9
markdown-it-py                    4.0.0
MarkupSafe                        3.0.3
math-verify                       0.5.2
matplotlib                        3.10.7
matplotlib-inline                 0.1.7
mdurl                             0.1.2
mistral_common                    1.8.5
mmengine-lite                     0.10.7
modelscope                        1.31.0
mpi4py                            4.1.1
mpmath                            1.3.0
ms-opencompass                    0.1.6
ms_swift                          3.9.1
ms-vlmeval                        0.0.18
msgpack                           1.1.2
msgspec                           0.19.0
multidict                         6.7.0
multiprocess                      0.70.16
networkx                          3.5
ninja                             1.13.0
nltk                              3.9.2
numba                             0.61.2
numpy                             2.2.6
nvidia-cublas-cu12                12.8.4.1
nvidia-cuda-cupti-cu12            12.8.90
nvidia-cuda-nvrtc-cu12            12.8.93
nvidia-cuda-runtime-cu12          12.8.90
nvidia-cudnn-cu12                 9.10.2.21
nvidia-cufft-cu12                 11.3.3.83
nvidia-cufile-cu12                1.13.1.3
nvidia-curand-cu12                10.3.9.90
nvidia-cusolver-cu12              11.7.3.90
nvidia-cusparse-cu12              12.5.8.93
nvidia-cusparselt-cu12            0.7.1
nvidia-ml-py                      13.580.82
nvidia-nccl-cu12                  2.27.3
nvidia-nvjitlink-cu12             12.8.93
nvidia-nvshmem-cu12               3.3.20
nvidia-nvtx-cu12                  12.8.90
openai                            2.5.0
openai-harmony                    0.0.4
OpenCC                            1.1.9
opencv-python                     4.11.0.86
opencv-python-headless            4.12.0.88
openpyxl                          3.1.5
orjson                            3.11.3
oss2                              2.19.1
outlines_core                     0.2.11
overrides                         7.7.0
packaging                         25.0
pandas                            2.3.3
parso                             0.8.5
partial-json-parser               0.2.1.1.post6
peft                              0.17.1
pexpect                           4.9.0
pillow                            11.3.0
pip                               25.2
platformdirs                      4.5.0
portalocker                       3.2.0
prettytable                       3.16.0
prometheus_client                 0.23.1
prometheus-fastapi-instrumentator 7.1.0
prompt_toolkit                    3.0.52
propcache                         0.4.1
protobuf                          6.33.0
psutil                            7.1.1
ptyprocess                        0.7.0
pure_eval                         0.2.3
py-cpuinfo                        9.0.0
pyarrow                           21.0.0
pyasn1                            0.6.1
pyasn1_modules                    0.4.2
pybase64                          1.4.2
pycountry                         24.6.1
pycparser                         2.23
pycryptodome                      3.23.0
pydantic                          2.11.10
pydantic_core                     2.33.2
pydantic-extra-types              2.10.6
pydub                             0.25.1
pyecharts                         2.0.9
Pygments                          2.19.2
pyparsing                         3.2.5
pypinyin                          0.55.0
python-dateutil                   2.9.0.post0
python-dotenv                     1.1.1
python-json-logger                4.0.0
python-Levenshtein                0.27.1
python-multipart                  0.0.20
pytz                              2025.2
PyYAML                            6.0.3
pyzmq                             27.1.0
qwen-vl-utils                     0.0.14
rank-bm25                         0.2.2
RapidFuzz                         3.14.1
ray                               2.50.1
referencing                       0.37.0
regex                             2025.9.18
requests                          2.32.5
rich                              13.9.4
rich-toolkit                      0.15.1
rignore                           0.7.1
rouge                             1.0.1
rouge-chinese                     1.0.3
rouge_score                       0.1.2
rpds-py                           0.27.1
rsa                               4.9.1
ruff                              0.14.1
s3transfer                        0.14.0
sacrebleu                         2.5.1
safehttpx                         0.1.6
safetensors                       0.6.2
scikit-learn                      1.7.2
scipy                             1.16.2
seaborn                           0.13.2
semantic-version                  2.10.0
sentence-transformers             5.1.1
sentencepiece                     0.2.1
sentry-sdk                        2.42.0
setproctitle                      1.3.7
setuptools                        79.0.1
shellingham                       1.5.4
simplejson                        3.20.2
six                               1.17.0
sniffio                           1.3.1
sortedcontainers                  2.4.0
soundfile                         0.13.1
soxr                              1.0.0
stack-data                        0.6.3
starlette                         0.48.0
sty                               1.0.6
swanlab                           0.6.12
sympy                             1.14.0
tabulate                          0.9.0
tenacity                          9.1.2
tensorboard                       2.20.0
tensorboard-data-server           0.7.2
termcolor                         3.1.0
threadpoolctl                     3.6.0
tiktoken                          0.12.0
timeout-decorator                 0.5.0
timm                              1.0.20
tokenizers                        0.22.1
tomlkit                           0.13.3
torch                             2.8.0
torchaudio                        2.8.0
torchvision                       0.23.0
tqdm                              4.67.1
traitlets                         5.14.3
transformers                      4.57.1
transformers-stream-generator     0.0.5
triton                            3.4.0
trl                               0.23.1
typer                             0.19.2
typing_extensions                 4.15.0
typing-inspection                 0.4.2
tzdata                            2025.2
urllib3                           2.5.0
uvicorn                           0.38.0
uvloop                            0.22.1
validators                        0.35.0
vllm                              0.11.0
watchfiles                        1.1.1
wcwidth                           0.2.14
websockets                        15.0.1
Werkzeug                          3.1.3
wheel                             0.45.1
word2number                       1.1
wrapt                             2.0.0
xformers                          0.0.32.post1
xgrammar                          0.1.25
xlsxwriter                        3.2.9
xxhash                            3.6.0
yapf                              0.43.0
yarl                              1.22.0
zipp                              3.23.0
zstandard                         0.25.0

**Comment by zhangzaibin** [comment]
vllm version 0.11.0 swift version 3.9.1

**Comment by zhangzaibin** [comment]
feel free to reopen it
